src/VisualStudio/Core/Def/Implementation/ProjectSystem/FileChangeWatcherProvider.cs

```diff
@@ -16,37 +18,32 @@ namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem
     [Export(typeof(FileChangeWatcherProvider))]
     internal sealed class FileChangeWatcherProvider
     {
-        private readonly TaskCompletionSource<IVsAsyncFileChangeEx> _fileChangeService = new(TaskCreationOptions.RunContinuationsAsynchronously);
-
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public FileChangeWatcherProvider(IThreadingContext threadingContext, [Import(typeof(SVsServiceProvider))] Shell.IAsyncServiceProvider serviceProvider)
+        public FileChangeWatcherProvider(
+            IThreadingContext threadingContext,
+            IAsynchronousOperationListenerProvider listenerProvider,
+            [Import(typeof(SVsServiceProvider))] Shell.IAsyncServiceProvider serviceProvider)
         {
-            // We do not want background work to implicitly block on the availability of the SVsFileChangeEx to avoid any deadlock risk,
-            // since the first fetch for a file watcher might end up happening on the background.
-            Watcher = new FileChangeWatcher(_fileChangeService.Task);
-
-            System.Threading.Tasks.Task.Run(async () =>
+            var fileChangeService = Task.Factory.StartNew(
```

Comment:
why StartNew, instead of Task.Run here.  ALternatively, could this not just call a normal async method that doesn't a `await TaskScheduler.Default` in it?  I ask primarily because i'm not a fan of added usages of Task.Factory.

Reply:
➡️ I can't pass `RunContinuationsAsynchronously` to `Task.Run`.